### PR TITLE
Red versus Blue: Civil war edition

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

As a loyalist, you will be able to chose blue, teal and purple

As a rebel, you can chose between red, orange and yellow

This only applies to greyscaled items (jaeger) , so I'll maybe remove PAS from HvH if needed

## Why It's Good For The Game

More visual clues to know who is who

## Changelog
:cl:
expansion: Forces HvH faction to wear certain colors
/:cl:

### CODE REVIEW

File: code/modules/unit_tests/spawn_humans.dm [hunk 1]
```diff
@@ -15,4 +15,4 @@
 		H.equipOutfit(outfits_to_test[outfits_to_test.len])
 		outfits_to_test.len--
 
-	sleep(20)
+	sleep(100)
```
Comment (BraveMole) [inline code/modules/unit_tests/spawn_humans.dm:18]: go figure

**Comment by SplinterGP** [comment]
COUNTER TERRORIST WINS!

**Comment by ilikesaying** [comment]
Caboose, Tucker and Doc

Sarge, Grif and Sister

File: code/__DEFINES/loadout.dm [hunk 2]
```diff
@@ -347,12 +347,12 @@ GLOBAL_LIST_INIT(medic_clothes_listed_products, list(
...
 		/obj/effect/essentials_set/basic_smartgunner = list(CAT_STD, "Standard kit", 0, "white"),
 		/obj/effect/essentials_set/basic_smartgunnermodular = list(CAT_STD, "Essential Jaeger Kit", 0, "white"),
-		/obj/effect/essentials_set/modular/skirmisher = list(CAT_AMR, "Light Skirmisher Jaeger kit", 0, "orange"),
-		/obj/effect/essentials_set/modular/scout = list(CAT_AMR, "Light Scout Jaeger kit", 0, "orange"),
-		/obj/effect/essentials_set/modular/infantry = list(CAT_AMR, "Medium Infantry Jaeger kit", 0, "orange"),
-		/obj/effect/essentials_set/modular/eva = list(CAT_AMR, "Medium EVA Jaeger kit", 0, "orange"),
-		/obj/effect/essentials_set/modular/assault = list(CAT_AMR, "Heavy Assault Jaeger kit", 0, "orange"),
-		/obj/effect/essentials_set/modular/eod = list(CAT_AMR, "Heavy EOD Jaeger kit", 0, "orange"),
+		/obj/effect/modular_set/skirmisher = list(CAT_AMR, "Light Skirmisher Jaeger kit", 0, "orange"),
+		/obj/effect/modular_set/scout = list(CAT_AMR, "Light Scout Jaeger kit", 0, "orange"),
```
Comment (TiviPlus) [inline code/__DEFINES/loadout.dm:351]: Explain this pls

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 3]
```diff
@@ -941,7 +979,26 @@ GLOBAL_LIST_INIT(available_specialist_sets, list("Scout Set", "Sniper Set", "Dem
...
+	for(var/typepath in spawned_gear_list)
+		var/item = new typepath(loc)
+		if(faction)
+			if(ismodulararmorarmorpiece(item))
+				var/obj/item/armor_module/armor/armorpiece = item
+				armorpiece.limit_colorable_colors(faction)
+				continue
+			if(ismodularhelmet(item))
+				var/obj/item/clothing/head/modular/helmet = item
+				helmet.limit_colorable_colors(faction)
```
Comment (TiviPlus) [inline code/game/objects/machinery/vending/new_marine_vendors.dm:997]: early continue/breaks

File: code/modules/modular_armor/armor_pieces.dm [hunk 4]
```diff
@@ -21,7 +21,7 @@
 	greyscale_colors = "#444732"
 
 	///Assoc list of color-hex for colors we're allowed to color this armor
-	var/static/list/colorable_colors = list(
+	var/list/colorable_colors = list(
```
Comment (TiviPlus) [inline code/modules/modular_armor/armor_pieces.dm:24]: Im deleting this in the color pr

File: code/modules/modular_armor/modular.dm [hunk 5]
```diff
@@ -362,7 +362,7 @@
 	var/visor_greyscale_config = /datum/greyscale_config/modular_helmet_visor_emissive
 
 	///Assoc list of color-hex for colors we're allowed to color this armor
-	var/static/list/colorable_colors = list(
+	var/list/colorable_colors = list(
 		"black" = "#474A50",
```
Comment (TiviPlus) [inline code/modules/modular_armor/modular.dm:366]: ditto

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by lKiseki** [comment]
Visual changes like these are most welcome since they help setup a clear distinction between both sides. We've discussed HvH before amongst admins and most of us agree we'd rather have color distinctions like these before doing another Civil War event, since it'd help greatly with many issues regarding the current state of the game mode.

Bravemole asked me to offer my input (assuming that's what he wanted?), and I gotta say I love this.

File: code/__DEFINES/loadout.dm (hunk 2 above)
Comment (BraveMole) [reply code/__DEFINES/loadout.dm:351]: essentials_set create the items in the init.

**Comment by henriquekill9576** [comment]
I'm my opinion I think the colors should be just blue and red, it might get a bit too confusing with all the 6 colors

**Comment by BraveMole** [comment]
> I'm my opinion I think the colors should be just blue and red, it might get a bit too confusing with all the 6 colors

I'll keep that in mind, it won't be hard to do that in the futur if people find the multiple colors too confusing

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 6]
```diff
@@ -941,7 +979,27 @@ GLOBAL_LIST_INIT(available_specialist_sets, list("Scout Set", "Sniper Set", "Dem
...
+/obj/effect/modular_set
+	///List of all gear to spawn
+	var/list/spawned_gear_list = list()
+
+/obj/effect/modular_set/Initialize(mapload, faction)
+	. = ..()
+	for(var/typepath in spawned_gear_list)
+		var/item = new typepath(loc)
+		if(!faction)
+			break
```
Comment (BraveMole) [inline code/game/objects/machinery/vending/new_marine_vendors.dm:991]:
```suggestion
			continue
```

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**